Without this change TotalView will not operate properly with 
SLURM launched jobs. We want the tasks to be in a STOPPED 
state and detached from the slurmd daemon. The default 
code leaves an extra SIGSTOP pending. Any time the task is 
allowed to process pending signals, it will be stopped again
and confuse TotalView.
===============================================================
--- ptrace.c    18 Sep 2003 23:50:00 -0000      1.3.2.1.4.1
+++ ptrace.c    22 Mar 2004 19:58:29 -0000
@@ -113,7 +113,10 @@ int ptrace_attach(struct task_struct *ta
        __ptrace_link(task, current);
        write_unlock_irq(&tasklist_lock);

-       force_sig_specific(SIGSTOP, task);
+       if (task->state != TASK_STOPPED)
+               force_sig_specific(SIGSTOP, task);
+       else
+               task->exit_code = SIGSTOP;
        return 0;

 bad:
===============================================================
